internal/services/automation/automation_variables_data_source.go

```diff
@@ -0,0 +1,289 @@
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"int": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeInt,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"null": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"string": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					// How do I make this related to the specific type?
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:         pluginsdk.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+						},
+
+						"description": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+
+						"encrypted": {
+							Type:     pluginsdk.TypeBool,
+							Computed: true,
+						},
+
+						"value": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceAutomationVariablesRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).Automation.VariableClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	automationId := variable.NewAutomationAccountID(subscriptionId, d.Get("resource_group_name").(string), d.Get("automation_account_name").(string))
+
+	variableList, err := client.ListByAutomationAccountComplete(ctx, automationId)
+	if err != nil {
+		return fmt.Errorf("listing variables in Automation Account %q: %+v", automationId, err)
```

Comment:
Resource IDs in the provider have a string method which prints out the type of resource, in which resource group etc. so we don't need to repeat that in the error message.
```suggestion
		return fmt.Errorf("listing variables in %s: %+v", automationId, err)
```